internal/logger/logger.go

```diff
@@ -114,26 +136,37 @@ func (f *loggerFactory) newLogger(level, prefix string) *log.Logger {
 
 func (f *loggerFactory) writer(level string) io.Writer {
 	if f.file != nil {
-		switch f.format {
-		case "json":
+		if f.format == "json" {
 			return &jsonWriter{
 				w:     f.file,
 				level: level,
 			}
-		case "text":
+		} else {
 			return &textWriter{
 				w:     f.file,
 				level: level,
 			}
 		}
-	}
-
-	switch level {
-	case "NOTICE":
-		return daemonize.StatusWriter
-	case "ERROR":
-		return os.Stderr
-	default:
-		return os.Stdout
+	} else if f.sysWriter != nil {
+		if f.format == "json" {
+			return &jsonWriter{
+				w:     f.sysWriter,
+				level: level,
+			}
+		} else {
+			return &textWriter{
+				w:     f.sysWriter,
+				level: level,
+			}
+		}
+	} else {
```

Comment:
Earlier this is not an else case, why did you make it else?

Reply:
This method (https://github.com/GoogleCloudPlatform/gcsfuse/blob/adeea67dbd30253431c9a052e2efb44d4a3c7996/internal/logger/logger.go#L53) is called in the start, at that time both file/syslog writers are not initialised and flow goes to the else block.

Reply:
you have return statement in every clause. you dont need else conditions.

if f.file != nil {
    return
}

if f.syswriter != nil {
   return
}

switch level